apis/core/v1beta1/openstackcontrolplane_webhook.go

```diff
@@ -46,14 +49,42 @@ var _ webhook.Validator = &OpenStackControlPlane{}
 func (r *OpenStackControlPlane) ValidateCreate() error {
 	openstackcontrolplanelog.Info("validate create", "name", r.Name)
 
-	return r.ValidateServices()
+	var allErrs field.ErrorList
+
+	allErrs = append(allErrs, r.ValidateServices()...)
+
+	if len(allErrs) != 0 {
+		return apierrors.NewInvalid(
+			schema.GroupKind{
+				Group: "openstackcontrolplane.openstack.org",
+				Kind:  "OpenstackControlplane",
+			},
+			r.Name, allErrs)
+	}
+
+	return nil
 }
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
 func (r *OpenStackControlPlane) ValidateUpdate(old runtime.Object) error {
 	openstackcontrolplanelog.Info("validate update", "name", r.Name)
 
-	return r.ValidateServices()
+	var allErrs field.ErrorList
+
+	if err := r.ValidateServices(); err != nil {
+		allErrs = append(allErrs, err...)
+	}
```

Comment:
nit: Create and update has different syntax , here `err` is a `field.ErrorList`. We can probably sync that if you're updating again.